__tests__/schema/content-generation.ts

```diff
@@ -0,0 +1,125 @@
+import gql from 'graphql-tag'
+import { rest } from 'msw'
+
+import { user as baseUser } from '../../__fixtures__'
+import { Client, given, nextUuid } from '../__utils__'
+
+const mockPythonServiceResponse = JSON.stringify({
+  heading: 'Exercises for 7th grade',
+  subtasks: [
+    {
+      question: 'What is the 2nd binomial formula?',
+    },
+  ],
+})
+
+// server is a global variable that is defined in __config__/setup.ts
+server.use(
+  rest.get(
+    `http://${process.env.CONTENT_GENERATION_SERVICE_HOST}/exercises`,
+    (req, res, ctx) => {
+      return res(ctx.status(200), ctx.text(mockPythonServiceResponse))
+    },
+  ),
+)
+
+const user = { ...baseUser, roles: ['de_reviewer'] }
+
+const userWithNoPermissionId = nextUuid(baseUser.id)
+const userWithNoPermission = {
+  ...baseUser,
+  id: userWithNoPermissionId,
+  roles: [],
+}
+
+const userWithWrongRole = {
+  ...baseUser,
+  id: nextUuid(userWithNoPermissionId),
+  // Being a guest or architect is not sufficient. One has to be a reviewer.
+  roles: ['guest', 'de_architect'],
+}
```

Comment:
```suggestion
const userWithWrongRole = {
  ...baseUser,
  id: nextUuid(baseUser),
  roles: ['login', 'de_architect'],
}
```
role `guest` is actually never used, I don't know why